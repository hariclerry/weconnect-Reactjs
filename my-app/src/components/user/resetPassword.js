import React from "react";

import NavBar from "components/common/navBar";
import Footer from "components/common/footer";
import "static/css/bootstrap.min.css";

const ResetUserPassword = ({Userpassword}) => (
  <div>
    <NavBar />

    <div className="container">
      <h3> You can now reset your password</h3>
    </div>
    <hr />
    <div style={{ margin: "200px 400px" }}>
    <form onSubmit={Userpassword}>
        <div className="form-group">
          <label htmlFor="email">Email:</label>
          <input
            type="email"
            className="form-control"
            name="email"
            id="email"
            placeholder="Enter email"
            required
          />
        </div>
        <div className="form-group">
          <label htmlFor="pwd"> New Password:</label>
          <input
            type="password"
            className="form-control"
            name="new_password"
            id="pwd"
            placeholder="Enter password"
            required
          />
        </div>
        <div className="form-group form-group-btn">
          <button
            type="submit"
            className="btn btn-success btn-lg"
            style={{ marginTop: "3%" }}
          >
            Reset Password
          </button>
        </div>
        <div className="clearfix" />
      </form>
    </div>
    <Footer />
  </div>
);

export default ResetUserPassword;
