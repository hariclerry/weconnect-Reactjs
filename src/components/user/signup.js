import React from "react";

import Notifications from "react-notify-toast";

import AuthNavbar from "components/common/authNavbar";
import Footer from "components/common/footer";
import "static/css/bootstrap.min.css";
import "static/css/jumbotron.css";
import "static/css/styles.css";


const UserRegistration = ({ signUp }) => (
 

      <div>
      <Notifications options={{zIndex: 200, top: '50px'}}/>
        <AuthNavbar />

        <div className="container">
          <h3> Please register to login</h3>
        </div>
        <hr />
        <div className="container" id="registration-form">
          <div className="image" />
          <div className="frm">
            <h1>Sign up</h1>
            <form onSubmit={signUp}>
              <div className="form-group">
                <label htmlFor="username">Username:</label>
                <input
                  type="text"
                  className="form-control"
                  name="name"
                  id="username"
                  placeholder="Enter username"
                required/>
              </div>
              <div className="form-group">
                <label htmlFor="email">Email:</label>
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  id="email"
                  placeholder="Enter email"
                required/>
              </div>
              <div className="form-group">
                <label htmlFor="pwd">Password:</label>
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  id="pwd"
                  placeholder="Enter password"
                required/>
              </div>
              <div className="form-group" className="fixed">
                <button type="submit" className="btn btn-success btn-lg">
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );

    export default UserRegistration;
 
