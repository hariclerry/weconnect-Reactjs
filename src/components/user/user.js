import React from "react";

import { Link } from "react-router-dom";
import Notifications from "react-notify-toast";

import NavBar from "components/common/navBar";
import Footer from "components/common/footer";
import TableSet from "../common/tableSet";
import "static/css/bootstrap.min.css";
import "static/css/fix-footer.css";

const Dashboard = ({ businesses, username }) => (
  <div>
    <Notifications />
    <NavBar />
    <div className="container">
      <div>
        <h4>
          Hello {username}, welcome to your dashboard
        </h4>
      </div>
      <hr />
      <br />
      <div className="fixed-footer">
        <div className="table-responsive">
          <table className="table table-striped">
            <TableSet />
            <tbody>
              {!businesses ? (
                <div> No Businesses found </div>
              ) : (
                businesses.map((business, index) => (
                  <tr key={business.id}>
                    <td>
                      <Link
                        to={`/singlebusiness/${business.id}`}
                        style={{ textDecoration: "None" }}
                      >
                        {" "}
                        {business["name"]}
                      </Link>
                    </td>
                    <td>{business.category}</td>
                    <td>{business.location}</td>
                    <td>{business.description}</td>
                    <td>
                      <Link
                        to={`/addreview/${business.id}`}
                        className="btn btn-info"
                        role="button"
                        style={{ marginRight: "20px" }}
                      >
                        Review
                      </Link>
                    </td>
                  </tr>
                ))
              )}
            </tbody>
          </table>
        </div>
      </div>
    </div>

    <Footer />
  </div>
);

export default Dashboard;
