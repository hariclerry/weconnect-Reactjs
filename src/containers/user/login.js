import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import {notify} from "react-notify-toast";

import { loginUser } from "actions/userActions";
import { jsonStringify } from "utils/jsonHelper";
import Login from "components/user/login";


export class UserLogin extends React.Component {
      // componentWillMount() {
      //   notify.show("Successfully logged in", "success");
      // }


  Login = e => {
    e.preventDefault(); //o prevent the default link behavior of opening a new page
    let userCredential = {
      email: e.target.elements.email.value,
      password: e.target.elements.password.value
    };
    this.props
      .loginUser(jsonStringify(userCredential))
      .then(() => {
        this.props.history.push("/dashboard");
      })
      .catch(err => {
        // alert("Wrong email or password");
        notify.show("Wrong user password or email", "warning");
      });
  };
  render() {

    return (
      <div>
        <Login 
        Login={this.Login}/>
      </div>
      
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  user: state.auth.loginData
});

export default withRouter(
  connect(
    mapStateToProps,
    { loginUser }
  )(UserLogin)
);
