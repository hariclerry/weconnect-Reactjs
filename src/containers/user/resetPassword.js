import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import {notify} from 'react-notify-toast';

import { passwordReset } from "actions/userActions";
import { jsonStringify } from "utils/jsonHelper";
import ResetUserPassword from "components/user/resetPassword";
import { checkIfUserIsLoggedIn } from "actions/userActions";


export class ResetPassword extends React.Component {
      componentDidMount() {
        checkIfUserIsLoggedIn(this.props.email, this.props.history);
      }


  Userpassword = e => {
    e.preventDefault(); //o prevent the default link behavior of opening a new page
    let userCredential = {
      email: e.target.elements.email.value,
      new_password: e.target.elements.new_password.value
    };
    this.props
      .passwordReset(jsonStringify(userCredential))
      this.props.history.push("/login");
      // .then(() => {
      //   this.props.history.push("/login");
      // })
      // .catch(err => {
      //   // alert("Wrong email or password");
      //   notify.show("Wrong user password or email", "warning");
      // });
  };
  render() {

    return (
      <div>
        <ResetUserPassword 
        Userpassword={this.Userpassword}/>
      </div>
      
    );
  }
}

passwordReset.propTypes = {
  passwordReset: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  userPasswordReset: PropTypes.object.isRequired,
  
};

const mapStateToProps = state => ({
  userPasswordReset: state.auth.resetPasswordData,
  email: state.auth.loginData.email,
});

export default withRouter(
  connect(
    mapStateToProps,
    { passwordReset, checkIfUserIsLoggedIn }
  )(ResetPassword)
);
