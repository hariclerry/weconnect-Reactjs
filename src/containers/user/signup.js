import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Notifications, { notify } from "react-notify-toast";

import { jsonStringify } from "utils/jsonHelper";
import Signup from "components/user/signup";
import { registerUser } from "actions/userActions";

/**
 * User registration Container Component for registering users.
 *
 */

export class UserSignup extends React.Component {
  // Prevent default submit event trigger(synthetic event):
  signUp = e => {
    e.preventDefault();
    let userCredential = {
      username: e.target.elements.name.value,
      email: e.target.elements.email.value,
      password: e.target.elements.password.value
    };
    this.props.registerUser(jsonStringify(userCredential)).then(() => {
      this.props.history.push("/login");
      notify.show("Successfully registered, please login", "success");
    });
  };

  render() {
    return (
      <div>
        <Signup signUp={this.signUp} />
        <Notifications />
      </div>
    );
  }
}

// validate props
Signup.protoTypes = {
  registerUser: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  user: state.auth.signupMessage
});

export default withRouter(
  connect(
    mapStateToProps,
    { registerUser }
  )(UserSignup)
);
