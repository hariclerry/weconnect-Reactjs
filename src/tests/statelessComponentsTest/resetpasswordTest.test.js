import React from "react";
import {shallow, mount} from "enzyme";
import { ResetPassword } from "../../containers/user/resetPassword"



const promiseFn = () =>  new Promise((resolve, reject) =>{
  
});

describe('ResetPassword', ()=>{
  let wrapper;

  beforeEach(()=>{
    const history = {
      push: jest.fn
    }
    wrapper = shallow(<ResetPassword passwordReset={promiseFn} history={history} />)
  })

  it('should reset user password', ()=>{
    const e = {
      preventDefault: jest.fn,
      target:{
        elements:{
          email: {
            value: "test@tes.com"
          },
          new_password: {
            value: "test"
          }
        }
      }
    }
    
    const spy = jest.spyOn(wrapper.instance(), "Userpassword");
    expect(spy).toHaveBeenCalledTimes(0);
    wrapper.instance().Userpassword(e);
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledTimes(1);
  })

  it('should match snapshot', () =>{
    expect(wrapper).toMatchSnapshot();
  })
})

