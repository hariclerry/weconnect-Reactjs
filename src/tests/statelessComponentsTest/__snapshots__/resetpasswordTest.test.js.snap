// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should match snapshot 1`] = `
<div>
  <ResetUserPassword
    Userpassword={[Function]}
  />
</div>
`;
